src/librustc_typeck/astconv.rs

```diff
@@ -88,9 +92,16 @@ struct ConvertedBinding<'tcx> {
     span: Span,
 }
 
-struct ParamRange {
-    required: usize,
-    accepted: usize
+#[derive(PartialEq)]
+enum GenericArgPosition {
+    Datatype,
+    Function,
+    Method,
```

Comment:
Can we call this... `Type`, `Value`, `AssocValue`? (or just `Associated`?)

Reply:
If this is related to `x.foo::<...>()`, that's "method *call*" specifically.

Reply:
The "call" was implicit, because they were all in expressions ;) I'll change the names.